src/main/java/app/model/agents/MemoryGraph.java

```diff
@@ -0,0 +1,192 @@
+        }
+        else if(cardinalDirections.get("West").equals(cardinalMovement))
+        {
+            return eastWestAggregate(currentCell, "West");
+        }
+        else
+        {
+            throw new RuntimeException("No cell aggregate could be calculated");
+        }
+    }
+
+    //TODO Refactor aggregates depending on best discussed solution
+    public double northSouthAggregate(GraphCell currentCell, String direction)
+    {
+        Vector currentPosition = currentCell.getPosition();
+        Vector aggregateDirection = cardinalDirections.get(direction);
+
+        GraphCell c1 = getVertexAt(currentPosition.add(aggregateDirection));
+        Vector TPosition = currentPosition.add(aggregateDirection.scale(2));
+        GraphCell c2 = getVertexAt(TPosition);
+
+        GraphCell[] aggregateVertices =  {c1, c2};
+        return aggregatePheromoneValues(aggregateVertices);
+    }
+
+    public double eastWestAggregate(GraphCell currentCell, String direction)
+    {
+        Vector currentPosition = currentCell.getPosition();
+        Vector aggregateDirection = cardinalDirections.get(direction);
+
+        GraphCell c1 = getVertexAt(currentPosition.add(aggregateDirection));
+        Vector TPosition = currentPosition.add(aggregateDirection.scale(2));
+        GraphCell c2 = getVertexAt(TPosition);
+
+        GraphCell[] aggregateVertices =  {c1, c2};
+        return aggregatePheromoneValues(aggregateVertices);
+    }
+
+    public double aggregatePheromoneValues(GraphCell[] aggregateVertices)
+    {
+        double pheromoneSum = 0;
+        for(GraphCell vertex: aggregateVertices)
+        {
+            if(vertex != null)
+            {
+                pheromoneSum += vertex.getPheromone();
+            }
+        }
+        return pheromoneSum;
+    }
+
+    public GraphCell getVertexFromCurrent(GraphCell currentCell, String direction)
+    {
+        Vector currentPosition = currentCell.getPosition();
+        Vector neighbouringCellDirection = cardinalDirections.get(direction);
+        return getVertexAt(currentPosition.add(neighbouringCellDirection));
+    }
+
+    public void leaveVertex(Vector position, double pheromoneValue)
+    {
+        GraphCell cell = getVertexAt(position);
+        cell.setOccupied(false);
+        cell.updatePheromone(pheromoneValue);
+    }
+
+    public GraphCell getVertexAt(Vector position)
+    {
+        return vertices.get(keyGenerator(position));
+    }
+
+    public String keyGenerator(Vector position)
+    {
+        Vector centrePosition = determineVertexCentre(position);
+        return centrePosition.getX() + " " + centrePosition.getY();
+    }
+
+    public Vector determineVertexCentre(Vector position)
+    {
+        int x_centre = calculateDimensionCentre(position.getX());
+        int y_centre = calculateDimensionCentre(position.getY());
+
+        return new Vector(x_centre, y_centre);
+    }
+
+    public int calculateDimensionCentre(double axisPosition)
```

Comment:
Consider making any of these internal functions into private methods, makes the class cleaner to use for outside users